File: tests/fixtures/polyglot/sample.swift
protocol Pricer {
    func quote(id: String) -> Int
}

class PricingEngine: Pricer {
    let prices: [String: Int]

    init(prices: [String: Int]) {
        self.prices = prices
    }

    func quote(id: String) -> Int {
        return compute(id: id)
    }

    private func compute(id: String) -> Int {
        return prices.count
    }
}

struct OrderId {
    let value: String
}

func demo(prices: [String: Int]) -> Int {
    let engine = PricingEngine(prices: prices)
    return engine.quote(id: "x")
}

File: tests/fixtures/polyglot/sample.php
<?php
namespace Shop;

interface Pricer {
    public function quote(OrderId $id): int;
}

class PricingEngine implements Pricer {
    private array $prices;

    public function __construct(array $prices) {
        $this->prices = $prices;
    }

    public function quote(OrderId $id): int {
        if ($id->value === '') {
            throw new UnknownOrderException($id);
        }
        $total = $this->compute($id);
        return $total;
    }

    private function compute(OrderId $id): int {
        return count($this->prices);
    }
}

function demo(array $prices): int {
    $engine = new PricingEngine($prices);
    return $engine->quote(new OrderId('x'));
}
